src/sentry/static/sentry/app/utils/withRepositories.tsx

```diff
@@ -0,0 +1,78 @@
+import React from 'react';
+import Reflux from 'reflux';
+import createReactClass from 'create-react-class';
+
+import {Client} from 'app/api';
+import {Repository} from 'app/types';
+import getDisplayName from 'app/utils/getDisplayName';
+import RepositoryStore from 'app/stores/repositoryStore';
+import {getRepositories} from 'app/actionCreators/repositories';
+
+type DependentProps = {
+  api: Client;
+  orgSlug: string;
+  projectSlug: string;
+  releaseVersion: string;
+};
+
+type InjectedProps = {
+  repos?: Array<Repository>;
+  reposLoading?: boolean;
+  reposError?: Error;
+};
+
+const withRepositories = <P extends InjectedProps>(
+  WrappedComponent: React.ComponentType<P>
+) =>
+  createReactClass<
+    Omit<P, keyof InjectedProps> & Partial<InjectedProps> & DependentProps,
```

Comment:
Why do you need `Partial` if they are all already optional?

Reply:
The keys are not optional. The values are optional.